src/ToricVarieties/cohomCalg/VanishingSets/constructors.jl

```diff
@@ -0,0 +1,32 @@
+#########################
+# 1: The Julia type for affine semigroups
+#########################
+
+@attributes mutable struct ToricVanishingSet
+    toric_variety::AbstractNormalToricVariety
+    scs::Vector{ShiftedCone}
+    i::fmpz
+    function ToricVanishingSet(toric_variety::AbstractNormalToricVariety, scs::Vector{ShiftedCone}, i::fmpz)
+        ambient_dimension = unique([ambient_dim(sc) for sc in scs])
+        if (length(ambient_dimension) > 1)
+            throw(ArgumentError("The ambient dimensions of the shifted cones must all agree."))
+        end
+        if (ambient_dimension[1] != rank(picard_group(toric_variety)))
```

Comment:
I'd rather write this as follows, to avoid allocating a vector containing all the ambient dimensions:
```suggestion
        ambient_dimension = ambient_dim(first(scs))
        if !all(sc -> ambient_dim(sc) == ambient_dimension, scs)
            throw(ArgumentError("The ambient dimensions of the shifted cones must all agree."))
        end
        if ambient_dimension != rank(picard_group(toric_variety))
```

Reply:
Sounds good, thank you!